src/lib/comms/sol-netctl-impl-connman.c

```diff
@@ -1445,6 +1445,44 @@ static const sd_bus_vtable agent_vtable[] = {
 };
 
 SOL_API int
+sol_netctl_report_error(struct sol_netctl_service *service,
+    const enum sol_netctl_agent_error_type type)
+{
+    int r;
+    sd_bus_message *reply;
+    const char *interface;
+
+    SOL_NULL_CHECK(_ctx.agent, -EINVAL);
+    SOL_NULL_CHECK(_ctx.agent_msg, -EINVAL);
+
+    if (_ctx.auth_service != service) {
+        SOL_WRN("The connection is not the one being authenticated");
+        return -EINVAL;
+    }
+
+    if (type == SOL_NETCTL_AGENT_RETRY) {
+        interface = sd_bus_message_get_interface(_ctx.agent_msg);
```

Comment:
Are you sure that interface will NEVER be NULL ? Otherwise we can have a crash in the strcmp below.